lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+                the weights multiplied by the weights from the
+                one_dimensional_weights_cube. The collapsing_coord will be the
+                leading dimension on the output cube.
+        """
+        result = iris.cube.CubeList()
+        if (masked_weights_cube.coord(collapsing_coord) !=
+                one_dimensional_weights_cube.coord(collapsing_coord)):
+            message = ("The collapsing_coord {} does not match on "
+                       "masked_weights_cube and "
+                       "one_dimensional_weights_cube".format(collapsing_coord))
+            raise ValueError(message)
+        for masked_weight_slice, one_dimensional_weight in zip(
+                masked_weights_cube.slices_over(collapsing_coord),
+                one_dimensional_weights_cube.slices_over(collapsing_coord)):
+            masked_weight_slice.data = (
+                masked_weight_slice.data * one_dimensional_weight.data)
+            result.append(masked_weight_slice)
+        result = result.merge_cube()
+        return result
+
+    @staticmethod
+    def normalised_masked_weights(weights_cube, collapsing_coord):
+        """
+        Normalise spatial weights along dimension associated with the
+        collapsing_coord. If for a given point the sum of the weights along
+        the collapsing_coord is zero then the returned normalised weight for
+        that point will also be zero. This correspsonds to the case where
+        there is missing data for that point for all slices along the
+        collapsing_coord.
+
+        Args:
+            weights_cube (iris.cube.Cube):
+                A cube with spatial weights and any other leading dimension.
+                This cube must have a coordinate matching the name given by
+                collapsing_coord which corresponds to the dimension along
+                which the normalisation is needed.
+            collapsing_coord (string):
+                The string that will match to a coordinate in both input cube.
+                This coordinate corresponds to the dimension along which the
+                normalisation is needed.
+
+        Returns:
+            result (iris.cube.Cube):
+                A cube with the same dimensions as the input cube, but with
+                the weights normalised along the collapsing_coord dimension.
+                The collapsing_coord will be the leading dimension on the
+                output cube.
+        """
+        summed_weights = weights_cube.collapsed(
+            collapsing_coord, iris.analysis.SUM)
+        result = iris.cube.CubeList()
+        # Slice over collapsing_coord so the dimensions match.
+        for weight_slice in (
+                weights_cube.slices_over(collapsing_coord)):
+            # Only divide where the sum of weights are positive. Setting
+            # the out keyword args sets the default value for where
+            # the sum of the weights are less than zero.
```

Comment:
`zero or less`?  Or just `zero`?  (I can't see a case where this plugin would produce negative weights - at least not one that wouldn't indicate serious problems elsewhere.)